sklearn/datasets/samples_generator.py

```diff
@@ -28,8 +27,7 @@ def _generate_hypercube(samples, dimensions, rng):
     if dimensions > 30:
         return np.hstack([_generate_hypercube(samples, dimensions - 30, rng),
                           _generate_hypercube(samples, 30, rng)])
-    out = astype(sample_without_replacement(2 ** dimensions, samples,
-                                            random_state=rng),
+    out = astype(random_state.permutation(2 ** dimensions)[:samples],
```

Comment:
Yeah this calls it with ``N=302231454903657293676544``. Sooo... we either limit hypercube to a smaller dimension or we make ``auto`` default to our implementation in the very strong edge-cases.